copy/copy.go

```diff
@@ -862,6 +869,14 @@ func (ic *imageCopier) copyLayers(ctx context.Context) error {
 		}()
 
 		for i, srcLayer := range srcInfos {
+			if srcLayer.Annotations == nil {
+				srcLayer.Annotations = make(map[string]string)
+			}
+			// Add default labels containing information of the target image and layers.
+			srcLayer.Annotations[layerTargetDigest] = srcLayer.Digest.String()
```

Comment:
This seems suspect. The manifest pretty much already exists to contain digests; why duplicate that?

If the intent of that annotation is to contain the DiffID (uncompressed digest), that’s 1) undocumented, 2) not implemented that way.

Reply:
This layer digest is stored to the annotations for being passed to the c/storage's `Store`.
ALS accessed via c/storage's `Store` can know which layers should be shown to c/storage.
This is especially needed for registry-backed ALS (e.g. eStargz, zstd:chunked).

Reply:
Again, https://github.com/containers/storage/pull/795#discussion_r551319228 . `TryReusingBlob` already has an _authoritative_ value of this digest directly as `srcInfo.Digest`; it’s not necessary to pass another copy, and especially not in this way.

Reply:
> Again, containers/storage#795 (comment) . TryReusingBlob already has an authoritative value of this digest directly as srcInfo.Digest; it’s not necessary to pass another copy, and especially not in this way.

Fixed this.
Now `*c/image.storageImageDestination.TryReusingBlobWithRef()` recieves digest through `srcInfo.Digest`.